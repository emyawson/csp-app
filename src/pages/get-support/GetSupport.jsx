// Modules
import React, { Fragment, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import Layout from '../../components/layout/Layout.jsx';
import BCrumb from '../../components/Bcrumb/BCrumb.jsx';
import Headline from '../../components/headline/Headline.jsx';
import ContactCard from '../../components/contactCard/ContactCard.jsx';
import SelfServiceOption from '../../components/selfServiceOption/SelfServiceOption.jsx';
import imgSVG from '../../utils/rectangle.svg';
import {
  selectPages, selectSelectedTopic, selectLoadingPages, selectContactPage,
} from '../../store/selectors';
// styles
import LoadingWrapper from '../../components/loading/LoadingWrapper.jsx';

const HeadlineLoader = LoadingWrapper(Headline);
const SelfServiceOptionLoader = LoadingWrapper(SelfServiceOption);
const ContactCardLoader = LoadingWrapper(ContactCard);

const GetSupport = () => {
  // new code
  const loadingPages = useSelector(selectLoadingPages);
  const pages = useSelector(selectPages);
  const topicID = useSelector(selectSelectedTopic);
  const selectedContactID = useSelector(selectContactPage);
  const [selfService, setSelfService] = useState(false);
  const [selfServiceButton, setSelfServiceButton] = useState(false);
  const [listSelectedtopics, setListSelectedtopics] = useState([]);
  const [selectedContactPage, setSelectedContactPage] = useState({});

  useEffect(() => {
    if (pages) {
      setListSelectedtopics(() => pages.topics.find(({ id }) => id === topicID).topics);
      setSelectedContactPage(listSelectedtopics.find(({ id }) => id === selectedContactID));
    }

    if (selectedContactPage) {
      setSelfService(() => (!!selectedContactPage.recommendation));
    }

    if (selfService) {
      setSelfServiceButton(() => (!!selectedContactPage.recommendation.callToActionText));
    }
  }, [
    pages,
    selectedContactPage,
    selfService,
    selfServiceButton]);

  return (
    <Layout>
      <BCrumb />
      {selectedContactPage && (
        <Fragment>
          <HeadlineLoader
            isLoading={loadingPages}
            title={selectedContactPage.headline}
            description={selectedContactPage.body}
            loadingMarginTop="60px"
            loadingWidth="505px"
            loadingHeight="106px"
          />

          {/* recommendation Option */}
          {selectedContactPage.recommendation && (
            <Fragment key={selectedContactPage.recommendation.id}>
              <SelfServiceOptionLoader
                isLoading={loadingPages}
                headline={selectedContactPage.recommendation.headline}
                content={selectedContactPage.recommendation.body}
                buttonLabel={selectedContactPage.recommendation.callToActionText}
                buttonLink={selectedContactPage.recommendation.callToActionUrl}
                // imgCMS={imgSVG}
                imgCMS={selectedContactPage.recommendation.image}
              />
            </Fragment>
          )}
          {selectedContactPage.contactMethods
            ? selectedContactPage.contactMethods.map((item, index) => (
              <Fragment key={item.id}>
                <ContactCardLoader
                  isLoading={loadingPages}
                  id={item.id}
                  cardTitle={item.titleContact}
                  cardContent={item.bodyContact}
                  contactType={item.type}
                  contactLabel={item.buttonLabel}
                  index={index}
                  phoneNumber={item.phoneNumber}
                  standardHours={item.standardHours}
                  holidayHours={item.holidayHours}
                  imgCMS={item.image}
                  // imgCMS={imgSVG}
                  recommended={selfService}
                  recommendedButtonExists={selfServiceButton}
                  disclaimer={item.disclaimer}
                  buttonLink={item.callToActionUrl}
                />
              </Fragment>
            ))
            : ''}
        </Fragment>
      )}
    </Layout>
  );
};

GetSupport.propTypes = {
  location: PropTypes.object,
};

export default GetSupport;
