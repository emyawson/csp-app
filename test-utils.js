import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
