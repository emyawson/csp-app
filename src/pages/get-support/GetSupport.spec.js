import React from 'react';
// import { render, screen } from "@testing-library/react";
import GetSupport from './GetSupport.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';
import pagesMock from '../../services/mock/pagesMock';

const defaultState = {
  router: {},
  loading: {},
  pages: pagesMock,
  selectedTopic: '1L8HKugsJOw3bgoVI4txrc',
  selectedContactPage: '1SuaI8C7L4mQBQLrgFwsGr',
  breadcrumbPathname: [],
  errors: {},
};

const renderComponent = () => renderWithRouter(<GetSupport />, {}, defaultState);

test('renders <GetSupport/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();

  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
