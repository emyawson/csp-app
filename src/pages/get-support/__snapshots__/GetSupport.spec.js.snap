// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <GetSupport/> 1`] = `
<div>
  <main
    class="page-container"
  >
    <div
      class="page-container__inner"
    >
      <div
        class="page-container__main"
      >
        <div
          class="breadcrumb"
        >
          <nav
            aria-label="Breadcrumb"
            class="breadcrumb__nav"
            data-cytest="breadcrumb"
            role="navigation"
          >
            <ol
              class="breadcrumb__list"
              itemprop="breadcrumb"
              itemscope=""
              itemtype="http://schema.org/BreadcrumbList"
            >
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    IKEA
                  </span>
                </a>
                 
                <meta
                  content="1"
                  itemprop="position"
                />
              </li>
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl/customer-service"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    Klantenservice
                  </span>
                </a>
                 
                <meta
                  content="2"
                  itemprop="position"
                />
              </li>
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl/customer-service/support"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    We are here to help and you should be greatful
                  </span>
                </a>
                 
                <meta
                  content="3"
                  itemprop="position"
                />
              </li>
            </ol>
          </nav>
        </div>
        <div
          class="headline"
        >
          <h1
            class="headline__title"
            data-cytest="headline"
          >
            Get support with store information
          </h1>
          <p
            class="headline__description"
          >
            Get store information such as hours, location, and directions. And hope to see you there!
          </p>
        </div>
        <div
          class="selfservice"
        >
          <div>
            <div
              class="selfservice__label"
              data-cytest="recommended-label"
            >
              <p
                class="label"
              >
                Aanbevolen
              </p>
            </div>
            <h2
              class="selfservice__title"
              data-cytest="selfservice-headline"
            >
              Store information
            </h2>
            <div
              class="selfservice__description"
              data-cytest="selfservice-description"
            >
              <div
                class="richtext-editorstyle "
              >
                <p>
                  PH: IKEA stores comes in different sizes and with different flavors. From the IKEA stores you know of to our planning studios where you can plan your future dream kitchen to our city stores where you can pick up affordable home decorations and get help with online purchases.
                </p>
                <p>
                  Use the store locator to find your closet store, planning studio, city store or a pick up point for your home deliveries.
                </p>
              </div>
            </div>
            <button
              class="btn btn--primary selfservice__button"
              data-cytest="button-selfService"
              type="button"
            >
              <span
                class="btn__inner"
              >
                
                Find your local store
              </span>
            </button>
          </div>
        </div>
        <div
          class="separator"
        />
        <div
          class="section"
        >
          <h2
            class="section__title"
          >
            Nog hulp nodig?
          </h2>
          <div
            class="section__body"
          >
            Heb je een andere vraag of een probleem  over de beschikbaarheid van producten en vind je het antwoord niet? Neem contact op met de klantenservice, zij helpen je verder.
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card__title"
            data-cytest="contactcard-headline"
          >
            <h2>
              Chat with our chatbot
            </h2>
          </div>
          <div
            class="card__content-block"
          >
            <div
              class="card__description"
              data-cytest="contactcard-description"
            >
              <div
                class="richtext-editorstyle "
              >
                <p>
                  Meet Buddy, our new chatbot helping in customer service and returns. Buddy is trained to help with information about store opening hours, stock check and order status, to name a few topics.
                </p>
              </div>
            </div>
            <button
              aria-label="De klantenservice is momenteel gesloten"
              class="btn btn--secondary card__button"
              data-cytest="contactcard-button"
              type="button"
            >
              <span
                class="btn__inner"
              >
                
                Start chat
              </span>
            </button>
          </div>
        </div>
        <div
          class="separator"
        />
      </div>
    </div>
  </main>
</div>
`;
